builtin/logical/nomad/backend_test.go

```diff
@@ -300,3 +315,169 @@ func TestBackend_CredsCreateEnvVar(t *testing.T) {
 		t.Fatalf("resp is error: %v", resp.Error())
 	}
 }
+
+func TestBackend_max_token_length(t *testing.T) {
+	config := logical.TestBackendConfig()
+	config.StorageView = &logical.InmemStorage{}
+	b, err := Factory(context.Background(), config)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cleanup, connURL, connToken := prepareTestContainer(t)
+	defer cleanup()
+
+	testCases := []struct {
+		title           string
+		roleName        string
+		tokenLength     int
+		envLengthString string
+	}{
+		{
+			title: "Default",
+		},
+		{
+			title:       "ConfigOverride",
+			tokenLength: 64,
+		},
+		{
+			title:       "ConfigOverride-LongName",
+			roleName:    "testlongerrolenametoexceed64charsdddddddddddddddddddddddd",
+			tokenLength: 64,
+		},
+		{
+			title:    "ConfigOverride-LongName-notrim",
+			roleName: "testlongerrolenametoexceed64charsdddddddddddddddddddddddd",
+		},
+		{
+			title:           "ConfigOverride-LongName-envtrim",
+			roleName:        "testlongerrolenametoexceed64charsdddddddddddddddddddddddd",
+			envLengthString: "16",
```

Comment:
Some good additional test cases would be this envvar being out of range or not a valid integer.